pages/settings/SettingsLog.qml

```diff
@@ -132,6 +132,20 @@ Rectangle {
             font.pixelSize: 18
             font.family: MoneroComponents.Style.fontRegular.name
             text: qsTr("Daemon log") + translationManager.emptyString
+            themeTransition: false
+            onColorChanged: {
+                console.log("color changed")
+                var flickableContentYBefore = flickable.contentY
+                var daemonLogText = consoleArea.text
+                consoleArea.clear();
+                if (MoneroComponents.Style.blackTheme) {
+                    var daemonLogTextSwitchedTextColor = daemonLogText.replace(/#000000/g, '#ffffff');
+                } else {
+                    var daemonLogTextSwitchedTextColor = daemonLogText.replace(/#ffffff/g, '#000000');
+                }
+                consoleArea.append(daemonLogTextSwitchedTextColor);
+                flickable.contentY = flickableContentYBefore
```

Comment:
```suggestion
                if (MoneroComponents.Style.blackTheme) {                       
                    consoleArea.text = consoleArea.text.replace(/#000000/g, '#ffffff');   
                } else {                                                            
                    consoleArea.text = consoleArea.text.replace(/#ffffff/g, '#000000');   
                }                                                              
```

Reply:
I tried this approach before, but it doesn't work. After switching theme, text area scrolls up (scrollbar position doesn't persist) and every time you type a command (like 'status'), it scrolls up again. That's why I used .clean() and then .append() on consoleArea and reconfigured    flickable.contentY.

Reply:
Oh, I didn't test this. Will recheck.

Reply:
this fixes the scroll bar, but this bug is still present: every time you type a command (like 'status'), it scrolls up again.

Reply:
read this
https://doc.qt.io/qt-5/qml-qtquick-textedit.html#text-prop

> "The append(), insert() and remove() methods provide more fine-grained control and remarkably better performance for modifying especially large rich text content."

Reply:
This code is in `onColorChanged` which will rarely get called so performance does not matter here. Apart from that append, insert, remove being faster in not valid if you clear it first.

Reply:
> every time you type a command (like 'status'), it scrolls up again.

It does not for me. Did you test my exact code?

![loop](https://user-images.githubusercontent.com/7697454/121071937-dd9b4e80-c7d0-11eb-8cbc-7480227d3d34.gif)

Reply:
Yes, I used your code. Did you switch theme before typing status?

Reply:
![logtext](https://user-images.githubusercontent.com/45968869/121076324-91530d00-c7d6-11eb-82b4-1f766de01582.gif)
This is what happens when I type status, after switching theme.

Reply:
Yes, can reproduce now. Don't know a better solution than your code.

Reply:
yeah, I spent a long time trying to fix this, but this was the only solution I found.